back/app/uploaders/base_image_uploader.rb

```diff
@@ -83,25 +82,8 @@ def resize_to_fill_using_convert(img, target_width, target_height, gravity = 'Ce
     img.extent "#{target_width}x#{target_height}" if current_width != target_width || current_height != target_height
   end
 
-  # Auto-orient the image based on EXIF data, before we strip the metadata.
-  def auto_orient
-    manipulate! do |img|
-      img.auto_orient
-      img
-    end
-  end
-
-  # Strip the image of any profiles, comments or these PNG chunks: bKGD,cHRM,EXIF,gAMA,
-  # iCCP,iTXt,sRGB,tEXt,zCCP,zTXt,date.
-  # (https://imagemagick.org/script/command-line-options.php#strip)
-  #
-  # It is a bit heavy-handed and removes a lot of things, including colour profiles which
-  # can have an effect on the image rendering. It also recompresses the image. Overall, it
-  # seems to remove vibrancy from the image.
+  # Strip the image of EXIF metadata, except ICC color profile and orientation metadata.
   def strip
-    manipulate! do |img|
-      img.strip
-      img
-    end
+    system("exiftool -all= -tagsFromFile @ -icc_profile -orientation -overwrite_original #{Shellwords.escape(@file.path)}")
```

Comment:
Did you test it while using S3 as the storage backend?

Also, it would be great to check the output of the command and report it to Sentry if it fails. Otherwise, it might fail silently forever :)

Reply:
Not tested with s3. Good idea.